corehq/apps/api/tests.py

```diff
@@ -125,6 +126,56 @@ def single_endpoint(self, id):
                                                           resource_name=self.resource.Meta.resource_name,
                                                           pk=id))
 
+    def _api_url(self, url, username=None):
+        if 'api_key' in url:
+            return url
+
+        username = username or self.username
+        api_key = self.api_key.key
+        if username != self.username:
+            web_user = WebUser.get_by_username(username)
+            api_key, _ = ApiKey.objects.get_or_create(user=WebUser.get_django_user(web_user))
+            api_key = api_key.key
+
+        api_params = urlencode({'username': username, 'api_key': api_key})
+
+        if "?" in url:
+            api_url = "%s&%s" % (url, api_params)
+        else:
+            api_url = "%s?%s" % (url, api_params)
+        return api_url
+
+    def _assert_auth_get_resource(self, url, username=None, password=None):
+        username = username or self.username
+        password = password or self.password
+
+        # session based auth should fail
+        self.client.login(username=username, password=password)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 401)
+
+        # api_key auth should succeed, caller can check for expected code
+        api_url = self._api_url(url, username)
+        response = self.client.get(api_url)
+        return response
+
+    def _assert_auth_post_resource(self, url, post_data, content_type='application/json', method="POST"):
```

Comment:
nit, but splitting this into `_assert_auth_post_resource` and `_assert_auth_put_resource` would be about as clean as taking a method param (which is a little weird when "post" is in the same of the method)